src/components/frontend-engine/frontend-engine.tsx

```diff
@@ -76,38 +75,46 @@ const FrontendEngineInner = (props: IFrontendEngineProps) => {
 		return () => subscription.unsubscribe();
 	}, [watch]);
 
-	// ================================================
+	// =============================================================================
 	// HELPER FUNCTIONS
-	// ================================================
-	const handleOnSubmit = (data: any) => {
-		console.log(data);
-		onSubmit && onSubmit();
-	};
-
-	const formatFormId = () => {
-		let formId = "frontend-engine";
+	// =============================================================================
+	useImperativeHandle<Partial<IFrontendEngineRef>, Partial<IFrontendEngineRef>>(ref, () => ({
```

Comment:
note: these are for OneService use case
- need to check whether form is valid to toggle submit button
- the submit button is outside `FrontendEngine`, there is a need to call submit from outside the form